### ISSUE TITLE
Project still active and maintained?

### ISSUE THREAD

**Issue by ocvist** [FlexSearch/FlexSearch#59]
Is this project still active and maintained?

**Comment by seemantr** [comment]
Thanks for getting in touch. Unfortunately, the project is dead now. The project was always reliant on IKVM and when IKVM died this project didn't have much future.

I am still thinking about the next logical steps and there are few things that could be done. I am thinking about porting an open source engine to dot net. There are few options- maybe a restricted feature set but good speed. Also, considering looking at Lucene.net.

**Comment by archnae** [comment]
Lucene.Net v.3 is pretty much alive and (imho) much faster then IKVM-based Lucene, even if the feature set is limited. Lucene.Net v.4 is also functional. The API is not that much different from the Java original, so porting shouldn't be an issue. I've thought about forking this project to Lucene.Net a year ago, but for me F#/.NET Core were showstoppers back then :)

**Comment by seemantr** [comment]
@archnae thanks for weighing in. Do you think there is a demand for such a product? I was keen to take it forward but in last few years due to shift towards SaaS solutions a lot of my clients moved to ES or Azure search.

To upgrade it won’t be a lot of effort.

**Comment by archnae** [comment]
It's hard to compete with ES or RavenDB, now that containers are everywhere and CPU cycles are dirt cheap. Frankly, I don't think that a self-standing non-bloated index-and-search product has a bright commercial future - an easier-to-use API and smaller footprint cannot compete with "industry standard", whatever it means, and embeddable niche is occupied with plain Lucene.

Nowadays money are, I believe,  in more specialized areas - ETL, data hygiene, record matching/deduplication - see, for example, [Duke](https://github.com/larsga/Duke) or dozens of commercial deduplication add-ons for Dynamics CRM. Our project sits firmly in this cozy corner, but Lucene is the smallest moving part of it - 90% of the code is about data analysis and transformation.

**Comment by seemantr** [comment]
@archnae I agree with your assessment. FlexSearch is mostly used by my clients for things like address searching and deduplication. That was the use case I wanted to extend. There is even a dedicated dedupe UI.

---
Few of the people following the project got in touch with me on my personal email saying that they have been using the product in Production for quite some time. They would like to see the product being maintained indefinitely. I really appreciate you getting in touch. It would be more useful if you feed your comments here on GitHub as it helps in understanding the size of the community behind an initiative. I won’t be able to respond to you on personal email.

After giving it a thought I will try to revive the project. My previous employer **Macmillan Cancer Support** __were quite open about contributing to Open Source and that is where this project really flourished__. I am talking with my current employer if I could contribute to Open Source projects in my spare time (Yes, I understand in this day and age). If they agree then I will announce a plan with a roadmap to take this project forward.

**Comment by archnae** [comment]
A bit of advice about (addresses, personal data etc) deduplication, based on 3 years of doing exactly that: the devil is in the handling of incomplete, broken or non-hierarchical addresses. The Lucene _search_ score isn't the best (or even good) _match_ score. For production quality/performance you need different per-attribute transform pipelines for lookup and matching, custom per-attribute comparators and a bit of background in statistical learning. You can hack it all on on top of ES and/or Lucene engine, but rebuilding/redeploying the project after every matching rules tweak is not sustainable, so it needs reasonably smart configuration system... very soon Lucene (or ES or Solr or RavenDB) is the least of your problems, at least in production use.
